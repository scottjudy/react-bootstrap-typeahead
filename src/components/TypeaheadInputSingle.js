// @flow

import React from 'react';

import Hint, { type ShouldSelect } from './Hint';
import Input from './Input';

import withClassNames from '../behaviors/classNames';

import type { RefCallback, ReferenceElement } from '../types';

type Props = {
  inputRef: RefCallback<HTMLInputElement>,
  referenceElementRef: RefCallback<ReferenceElement>,
  shouldSelectHint?: ShouldSelect,
};

export default withClassNames(
  ({ inputRef, referenceElementRef, shouldSelectHint, ...props }: Props) => (
    <Hint shouldSelect={shouldSelectHint}>
      <Input
        {...props}
        ref={(node) => {
          inputRef(node);
          referenceElementRef(node);
        }}
      />
    </Hint>
  )
);
