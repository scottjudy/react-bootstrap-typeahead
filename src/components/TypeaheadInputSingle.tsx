import React, { RefCallback } from 'react';

import Hint, { ShouldSelect } from './Hint';
import Input from './Input';

import withClassNames from '../behaviors/classNames';

interface TypeaheadInputSingleProps {
  inputRef: RefCallback<HTMLInputElement>;
  referenceElementRef: RefCallback<HTMLInputElement>;
  shouldSelectHint?: ShouldSelect;
}

export default withClassNames(
  ({
    inputRef,
    referenceElementRef,
    shouldSelectHint,
    ...props
  }: TypeaheadInputSingleProps) => (
    <Hint shouldSelect={shouldSelectHint}>
      <Input
        {...props}
        ref={(node) => {
          inputRef(node);
          referenceElementRef(node);
        }}
      />
    </Hint>
  )
);
