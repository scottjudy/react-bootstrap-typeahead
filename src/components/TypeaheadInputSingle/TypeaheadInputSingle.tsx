import React from 'react';

import Hint from '../Hint';
import Input from '../Input';

import { TypeaheadInputProps } from '../../types';
import { propsWithBsClassName } from '../../utils';

const TypeaheadInputSingle = ({
  inputRef,
  referenceElementRef,
  ...props
}: TypeaheadInputProps) => (
  <Hint>
    <Input
      {...propsWithBsClassName(props)}
      ref={(node) => {
        inputRef(node);
        referenceElementRef(node);
      }}
    />
  </Hint>
);

export default TypeaheadInputSingle;
