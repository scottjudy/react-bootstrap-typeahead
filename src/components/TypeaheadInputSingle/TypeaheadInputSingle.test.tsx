import * as stories from './TypeaheadInputSingle.stories';

import { generateSnapshots } from '../../tests/helpers';

describe('<TypeaheadInputSingle>', () => {
  generateSnapshots(stories);
});
