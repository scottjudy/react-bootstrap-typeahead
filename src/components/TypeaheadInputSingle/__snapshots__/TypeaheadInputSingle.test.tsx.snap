// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeaheadInputSingle> Default story renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    className="rbt-input-main form-control rbt-input"
    onKeyDown={[Function]}
    placeholder="This is a default input..."
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value=""
  />
</div>
`;

exports[`<TypeaheadInputSingle> Invalid story renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    className="rbt-input-main form-control rbt-input is-invalid"
    onKeyDown={[Function]}
    placeholder="This is an invalid input..."
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value=""
  />
</div>
`;

exports[`<TypeaheadInputSingle> Large story renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    className="rbt-input-main form-control rbt-input form-control-lg"
    onKeyDown={[Function]}
    placeholder="This is a large input..."
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value=""
  />
</div>
`;

exports[`<TypeaheadInputSingle> Small story renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    className="rbt-input-main form-control rbt-input form-control-sm"
    onKeyDown={[Function]}
    placeholder="This is a small input..."
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value=""
  />
</div>
`;

exports[`<TypeaheadInputSingle> Valid story renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    className="rbt-input-main form-control rbt-input is-valid"
    onKeyDown={[Function]}
    placeholder="This is a valid input..."
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value=""
  />
</div>
`;

exports[`<TypeaheadInputSingle> WithHint story renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "height": "100%",
      "position": "relative",
    }
  }
>
  <input
    className="rbt-input-main form-control rbt-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    value="Ca"
  />
  <input
    aria-hidden={true}
    className="rbt-input-hint"
    readOnly={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "boxShadow": "none",
        "color": "rgba(0, 0, 0, 0.54)",
        "left": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    tabIndex={-1}
    value="California"
  />
</div>
`;
